src/gfx/reverse.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+ * This file is part of RGBDS.
+ *
+ * Copyright (c) 2022, Eldred Habert and RGBDS contributors.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include "gfx/reverse.hpp"
+
+#include <algorithm>
+#include <array>
+#include <assert.h>
+#include <cinttypes>
+#include <errno.h>
+#include <fstream>
+#include <optional>
+#include <png.h>
+#include <string.h>
+#include <tuple>
+#include <vector>
+
+#include "defaultinitalloc.hpp"
+
+#include "gfx/main.hpp"
+
+static DefaultInitVec<uint8_t> readInto(std::string path) {
+	std::filebuf file;
+	file.open(path, std::ios::in | std::ios::binary);
+	DefaultInitVec<uint8_t> data(128 * 16); // Begin with some room pre-allocated
+
+	size_t curSize = 0;
+	for (;;) {
+		size_t oldSize = curSize;
+		curSize = data.size();
+
+		// Fill the new area ([oldSize; curSize[) with bytes
+		size_t nbRead =
+		    file.sgetn(reinterpret_cast<char *>(&data.data()[oldSize]), curSize - oldSize);
+		if (nbRead != curSize - oldSize) {
+			// Shrink the vector to discard bytes that weren't read
+			data.resize(oldSize + nbRead);
+			break;
+		}
+		// If the vector has some capacity left, use it; otherwise, double the current size
```

Comment:
Get the file size, reserve that many bytes in the vector, and read it all at once?

Reply:
But what if it's a pipe?